test/CarouselSpec.js

```diff
@@ -100,6 +100,18 @@ describe('<Carousel>', () => {
 
       wrapper.find('.carousel-indicators li').last().simulate('click');
     });
+
+    it(`should allow refs to be attached and expose next, prev functions`, () => {
+      let inputRef = React.createRef();
+      const wrapper = mount(
+        <Carousel ref={(input) => (inputRef = input)}>{items}</Carousel>,
```

Comment:
**issue:** this is not how refs are meant to work. you're meant to pass in the ref directly, then use `ref.current`.

also, if we're going to assert on the ref, we should probably assert that at least e.g. the `next` method actually works

Reply:
updated